### ISSUE TITLE
linter: `typescript/no-unnecessary-type-parameters` - false positive on classes

### ISSUE THREAD

**Issue by DetachHead** [oxc-project/oxc#25056]
### What version of Oxlint are you using?

1.75.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["typescript"],
  "rules": {
    "typescript/no-unnecessary-type-parameters": "error"
  }
}
```

### What happened?

```ts
declare class Box<T> { // error: Type parameter T is used only once in the class signature.
  value: T;
}
```

this rule is only relevant to functions, where a single use of a generic is no different from just using `unknown` (or `never` if used in a contravariant position). removing the generic changes the behavior of the class, for example:

```ts
declare const numberBox: Box<number>;
numberBox.value = 1;
numberBox.value = ""; // error
```

if the generic was removed, that error would go away.

the rule is correctly not reported for types and interfaces:

```ts
interface Bar<T> {
  value: T; // no error (correct)
}

type Baz<T> = {
  value: T; // no error (correct)
};
```

so it shouldn't be reported for classes either.

also see https://github.com/typescript-eslint/typescript-eslint/issues/9961
